.github/workflows/build.yml

```diff
@@ -0,0 +1,733 @@
+#
+# Copyright (c) 2021-present, Trail of Bits, Inc.
+# All rights reserved.
+#
+# This source code is licensed in accordance with the terms specified in
+# the LICENSE file found in the root directory of this source tree.
+#
+
+name: Build
+
+on:
+  # Run this workflow once every 6 hours against the master branch
+  schedule:
+   - cron: "0 */6 * * *"
+
+  push:
+    branches:
+      - 'master'
+
+    tags:
+      - '*'
+
+  pull_request:
+    branches:
+      - '*'
+
+
+
+
+jobs:
+  cleanup_stale_workflows:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Clone the anvill repository
+        uses: actions/checkout@v2
+
+      - name: Install Python dependencies
+        run: |
+          pip install requests
+
+      - name: Cancel old workflow runs
+
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+        run: |
+          python3 .github/scripts/abort_previous_workflows.py
+
+
+
+
+  build_linux:
+    needs: cleanup_stale_workflows
+
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          - { name: 'ubuntu', tag: '18.04' }
+          - { name: 'ubuntu', tag: '20.04' }
+
+        llvm: [ '9', '10', '11' ]
+        cxxcommon_version: [ 'v0.1.1' ]
+
+    runs-on: ubuntu-20.04
+    container:
+      image: docker.pkg.github.com/trailofbits/cxx-common/vcpkg-builder-${{ matrix.image.name }}:${{ matrix.image.tag }}
```

Comment:
Compatibility is tricky, ideally we would want to use something like the [osquery toolchain](https://github.com/osquery/osquery-toolchain) to pin the glibc version compatibility in order to guarantee that generated binaries will work on any distro. This would allow us to build once and deploy everywhere, like we are doing with osquery.

In the ideal case, a single commit from anvill would let use reproduce the build - but it's hard if we don't pin the cxx-common version